sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+  }
+}
+```
+
+Sometimes a paged result may contain multiple collections and you may want to iterate over all of them.
+
+```C++
+for (auto page = directoryClient.ListFilesAndDirectories(); page.HasPage(); page.MoveToNextPage())
+{
+  for (const auto& d : page.Directories)
+  {
+    std::cout << "Directory: " << d.Name << std::endl;
+  }
+  for (const auto& f : page.Files)
+  {
+    std::cout << "File: " << f.Name << std::endl;
+  }
+}
+```
+
+## Can I specify an API-version other than the default one?
+
+Yes, each client options takes an `ApiVersion` as optional parameter, with which you can specify an API-version used for all the HTTP requests from this client.
+Client spawned from another client instance will inherit the settings.
+
+```C++
+// serviceClient sends HTTP requests with default API-version, which will change as version evolves.
+auto serviceClient = BlobServiceClient::CreateFromConnectionString(GetConnectionString());
+
+// serviceClient sends HTTP requests with specified API-version.
+BlobClientOptions clientOptions;
+clientOptions.ApiVersion = "2019-12-12";
+auto serviceClient = BlobServiceClient::CreateFromConnectionString(GetConnectionString, clientOptions);
+// containerClient inherits this setting.
+auto containerClient = serviceClient.GetBlobContainerClient(containerName);
+```
+
+Please note that multiple-version is not supported yet by Storage C++ SDK (This may change in the future).
+Which means, the code above only changes the `x-ms-version` request header and doesn't change SDK behavior.
+SDK will still set headers and query parameters that are added between the customized API version (which is usually older) and latest API version,
+and still expect some headers or fields in HTTP response.
+Furthermore, this scenario is not covered by testing, although most of the APIs should be able to work. You should do thorough testing before your code goes to production.
+
+## How can I set a custom UserAgent string?
+
+We recommend you set an application ID with code below, so that it can be identified from which application, SDK, platform the request was sent. The information could be useful for troubleshooting and telemetry purposes.
+
+```C++
+BlobClientOptions clientOptions;
+clientOptions.Telemetry.ApplicationId = "SomeApplication v1.2.3";
+
+// The User-Agent string will be something like:
+// SomeApplication v1.2.3 azsdk-cpp-storage-blobs/12.9.0-beta.3 (Windows 10 Enterprise 6.3 19045 19041.1.amd64fre.vb_release.191206-1406)
+```
+
+## What encoding is used for input and output?
+
+All URLs should be URL-encoded, other resource names should be in UTF-8 encoding.
+This applies to both input variables and output.
+If your code runs in an environment where the default locale and encoding is not UTF-8, you should encode before passing variables into SDK and decode after reading a variable from SDK.
+
+```C++
+// containerUrl should be URL-encoded
+auto containerClient = BlobContainerClient(containerUrl);
```

Comment:
It's probably useful to show how the customer would Ur-encode the input in the code snippet.